libdnf/repo/solv_repo.cpp

```diff
@@ -312,7 +319,7 @@ void SolvRepo::rewrite_repo(libdnf::solv::IdQueue & fileprovides) {
 
     logger.debug("Rewriting repo \"{}\" with added file provides", config.get_id());
 
-    if (!config.build_cache().get_value() || main_nrepodata < 2 || fileprovides.size() == 0) {
+    if (!config.build_cache().get_value() || main_solvables_start == 0 || fileprovides.size() == 0) {
```

Comment:
I'm not sure what was the purpose of the `main_nrepodata < 2` check. I think it was just a check if the data were loaded at all, so I replaced it with an equivalent of that (the solvables don't start from 0, so if `main_solvables_start` is 0 we can be certain we haven't actually loaded the main data).